### DISCUSSION TITLE
Handling of ObservingCapability.observedProperty.href and procedure.href for WFD MonitoringSites

### DISCUSSION THREAD

**Discussion by JohannaOtt** [INSPIRE-MIF/helpdesk#127] [Q&A]
According to p. 34 of [this presentation](https://inspire.ec.europa.eu/sites/default/files/inspire_priority_data_sets_-_water_framework_directive_-_2020-06-09_-_fe.pdf), the values for `EnvironmentalMonitoringFacility.observingCapability.ObservingCapability.observedProperty.href` and `EnvironmentalMonitoringFacility.observingCapability.ObservingCapability.procedure.href` are "http://dd.eionet.europa.eu/vocabularyconcept/common/NilReasonType/template".

So basically, a nilReason value is provided, but not in the `nilReason` attribute, but in the `href` attribute of the elements. The codelist value is defined as "The value will be available later."

Is it planned to provide according values or are they maybe already available and the example in the presentation is not up to date anymore?
Or is it planned to stick with the nilReason-as-href-value approach over the longterm?

**Comment by sMorrone** [comment]
Dear Johanna,
the slide 34 of the presentation is related to flat WFD GML , which has an INSPIRE-based structure but **it is not INSPIRE compliant!** To demonstrate how it is simple to transform a WFD reporting delivery (flat GML ) into INSPIRE compliant data some examples are provided in the presentation and If you go to slide 32, for example, you can see that the nilreason is provided correctly as nilreason=’ https://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated’.
Lastly, the https://dd.eionet.europa.eu/vocabulary/common/NilReasonType is meant for the WFD reporting so it contains additional values not corresponding to any value in VoidReasonValue codelist.

Hope this clarifies
Stefania

**Comment by KathiSchleidt** [comment]
Dear Johanna,
actually, if the ObservingCapability contains neither the ObservedProperty nor the Procedure, I'd say it doesn't exist at all (what does it contain?). As ObsCaps are foreseen on EF to indicate what's being measured at this EF but are not mandatory in the EF model, thus at this level, you could just leave them out.
:)
Kathi

**Comment by JohannaOtt** [comment]
Dear Stefania, dear Kathi,

Thank you for your quick feedback, this is very valuable!

> slide 34 of the presentation is related to flat WFD GML , which has an INSPIRE-based structure but it is not INSPIRE compliant! 

Thanks for clarification. To me, a flat gml does not have any nested elements, so I expected this to be an exemple/best practice for WFD -> INSPIRE harmonisation.


> you can see that the nilreason is provided correctly as nilreason=’ https://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated’.

The problem in the case of `EnvironmentalMonitoringFacility.observingCapability.ObservingCapability.observedProperty.href` and `EnvironmentalMonitoringFacility.observingCapability.ObservingCapability.procedure.href` is not, that wrong nilReason values are used, but that a nilReason value is used as value for `href`. Both elements are not voidable - so I think, the "wrong" nilReason value is used consciously here in order to provide any value for the element so that it is valid. I imagined it to have a kind of placeholder function as the definition of the value linked is "The value will be available later." and that the href values will be replaced with meaningful values once they are available. Until then, the placeholder values can be used to be INSPIRE compliant.


> actually, if the ObservingCapability contains neither the ObservedProperty nor the Procedure, I'd say it doesn't exist at all (what does it contain?). As ObsCaps are foreseen on EF to indicate what's being measured at this EF but are not mandatory in the EF model, thus at this level, you could just leave them out.

I thought about doing so, too, but the the example-INSPIRE-alike gml in the presentation made me think that it is important for the WFD community to provide the GroundWaterBody as `featureOfInterest` in the `ObservableCapability` and they therefore decided to create that mostly empty ObservingCapability element. But I am totally fine with just not creating it and thus being INSPIRE compliant ;)


All the best

Johanna

**Comment by KathiSchleidt** [comment]
Hi Johanna,
taking a further look at the cardinalities on ObsCaps, I do worry a bit about totally leaving out the observedProperty and procedure (neither are voidable in the correct model, thus even if you're using an alternative encoding, it would be formally wrong against the INSPIRE models). 
An alternative approach that comes to mind is to provide a placeholder observedProperty and procedure referencing whatever is good practice under WFD, e.g. WFD_ObservedProperties and WFD_Procedures to at least indicate that data from these EFs will be collected in line with the WFD modalities
Would this work?
:)
Kathi

**Comment by JohannaOtt** [comment]
Hi Kathi,

> taking a further look at the cardinalities on ObsCaps, I do worry a bit about totally leaving out the observedProperty and procedure

I just left out the complete ObservingCapability element, not only observedProperty and procedure. As EnvironmentalMonitoringFacility.observingCapability has cardinality 0..n, this should be fine.


> An alternative approach that comes to mind is to provide a placeholder observedProperty and procedure referencing whatever is good practice under WFD, e.g. WFD_ObservedProperties and WFD_Procedures to at least indicate that data from these EFs will be collected in line with the WFD modalities

That is, what I thought the WFD community had done in the presentation I was referring to. Although the placeholder in that case was no observedProperty or procedure, but a more generic placeholder (that I thought, they would maybe be replaced at some point in time). So you are basically proposing to link to the codelist http://dd.eionet.europa.eu/vocabulary/wise/ObservedProperty/view without referring to a specific code? Unfortunately, there is no such codelist for procedure that could be used.


> Would this work?

I am not an expert in WFD data - so I cannot really tell if that would be a solution, the WFD community could agree on. I guess for now, I will just leave the ObservingCapability out to be on the safe side.


All the best

Johanna

**Comment by sMorrone** [comment]
Dear both,
I agree that the example in the presentation is aimed  to provide  placeholder _observedProperty_ and _procedure_. In my opinion, however, these should be replaced only by the actual specific codes when available (so for me it is a good choice not to provide the complete _ObservingCapability_ element in the INSPIRE GML if nothing is available for it). 
This discussion, however, highlights once again the need for a guide /an alignment between e-reporting deliveries and related encoding of the INSPIRE Priority datasets.
In the meantime, I will try to get better insights on this from the EEA.

Stefania

**Comment by KathiSchleidt** [comment]
What I do like about @JohannaOtt approach is that there is indication of the FoI, so one knows what this facility is monitoring. The real reason I checked the model today was to see if there was any other option for providing a link to the FoI (there isn't)

@sMorrone think you could lobby the EEA towards providing such placeholder entries for _observedProperty_ and _procedure_? While the actual values for these would be more informative, noting that "WFD Approved" indicates professional methodology vs. freestyle

**Comment by percursoaleatorio** [comment]
Dear all,
Your interpretations are correct: the _ObservingCapability_ was introduced in the WFD model (since 2015) so that a link between the monitoring site and the water body could be provided using the _featureOfInterest_. That relationship (between the site and the water body) is required in the WFD reporting.
However, the _observedProperty_ and _procedure_ are not requested (in the WFD reporting). Therefore the presentation only shows a "placeholder" for such values.

Stating it in another way: in a hypothetical scenario where the WFD spatial data strictly required for reporting purposes could be harvested from INSPIRE services,  _observedProperty_ and _procedure_ would be ignored, but _featureOfInterest_ would be harvested.

Hope this helps, néry

**Comment by JohannaOtt** [comment]
Hi Néry,

Thank you for the input. Would you think the solution Kathi proposed would be feasible then for the WFD community? This way, everybody would now what to put there and the information on the FoI would not get lost due to INSPIRE compliance.

> An alternative approach that comes to mind is to provide a placeholder observedProperty and procedure referencing whatever is good practice under WFD, e.g. WFD_ObservedProperties and WFD_Procedures to at least indicate that data from these EFs will be collected in line with the WFD modalities

> @sMorrone think you could lobby the EEA towards providing such placeholder entries for observedProperty and procedure? While the actual values for these would be more informative, noting that "WFD Approved" indicates professional methodology vs. freestyle


All the best

Johanna
